pyccel/codegen/printing/ccode.py

```diff
@@ -544,6 +544,14 @@ def stored_in_c_pointer(self, a):
             return False
         return a.is_pointer or a.is_optional
 
+    def create_tmp_var(self, init_val, match_var):
+        tmp_var_name = self._parser.get_new_name('tmp')
+        tmp_var = Variable(name = tmp_var_name, dtype = match_var.dtype)
+        self._additional_declare.append(tmp_var)
+        assign = Assign(tmp_var, init_val)
+        self._additional_code += self._print(assign) + '\n'
+        return VariableAddress(tmp_var)
```

Comment:
I think you should return a `Variable`, you don't know what this function will be used for in the future and it is possible that a `VariableAddress` is not required